zanataclient/zanatalib/projectutils.py

```diff
@@ -142,3 +144,82 @@ def dict2xml(tag, dict):
             child.text = str(val)
             elem.append(child)
         return ToolBox.prettify(elem)
+
+
+class FileMappingRule(object):
+    """
+    Build translation's paths for pull and push considering rules
+
+    The mapping rules configuration is optional in zanata.xml.
+    If not specified, standard rules are applied according to project type.
+    """
+
+    project_filemapping_default_config = {
+        'gettext': '{path}/{locale_with_underscore}.{extension}',
+        'podir': '{locale}/{path}/{filename}.{extension}',
```

Comment:
Ok. I'll try to take a look at this patch since I was planning to come back here (repo) and look at problems I've been having with podir project types as from my viewpoint it seems broken in this CLI  (but this is very much unrelated to this MR )